---
title: Merging HGDP and 1000 Genomes
output:
  html_document: default
---

In this document we will download HGDP dataset and merge it with 1000 Genomes to end up with a full reference sample file.
We load the already downloaded files.

```{r, warning=FALSE}
setwd('..')
path <- getwd()
setwd(paste(path, "/Results/Temp/hgdp", sep = ""))
vars <- read.table("HGDP_FinalReport_Forward.txt")
map  <- read.table("HGDP_Map.txt")
```

We now format the HGDP data file into .ped plink data format.
Specifically, the .ped file requires 6 columns (Family ID, Individual ID, Paternal ID, Maternal ID, Sex, Phenotype), followed by 2 columns for each variant (one for each allele).


```{r}
hgdp_ped       <- as.data.frame(matrix(nrow = ncol(vars), ncol = (6 + (2 * nrow(map)))))
hgdp_ped[,1]   <- colnames(vars)
hgdp_ped[,2]   <- colnames(vars)
hgdp_ped[,3:6] <- 0


for(colu in 1:ncol(vars)){
  count <- 1
  for(rows in 1:nrow(vars)){
    alleles  <- unlist(strsplit(as.character(vars[rows,colu]), split=''))
    hgdp_ped[colu, 6 + count]     <- alleles[1]
    hgdp_ped[colu, 6 + count + 1] <- alleles[2]
    count <- count + 2
  }
}

write.csv(hgdp_ped, file = "HGDP.ped", quote = FALSE, row.names = FALSE, col.names = FALSE)
```
